internal/services/search/search_service_resource_test.go

```diff
@@ -242,6 +242,64 @@ func TestAccSearchService_identity(t *testing.T) {
 	})
 }
 
+func TestAccSearchService_identityUserAssigned(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_search_service", "test")
+	r := SearchServiceResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data, "standard"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.identityUserAssigned(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data, "standard"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSearchService_identitySystemAndUserAssigned(t *testing.T) {
```

Comment:
Can we change these to being a single create, and then add an `_update` that cycles through the configs?  This then tests the create time inclusion of the identity block of each type, and covers the updating of it. 
so we'd have:
* `TestAccSearchService_identityUserAssigned` - just using `Config: r.identityUserAssigned(data),` and the `data.ImportStep()`
* `TestAccSearchService_identitySystemAndUserAssigned` - just using `r.identitySystemAndUserAssigned(data)` and `data.ImportStep()` 
* a new `TestAccSearchService_identityUpdate` test that runs through a basic -> UserAssigned -> UserAssignedSystemAssigned -> basic lifecycle

Reply:
thanks @jackofallops for your review and feedback!
I updated the tests.
The update test cycles: basic -> identity (which is system assigned) -> user assigned -> system and user assigned -> basic.

One side question: 
Could/Should I have done this api update differently? 
I read I'm supposed to split api updates and new features. After the api update here the type of the returned object changed to I was thinking its ok to do that change as well. The alternative would have been to map / cast back to only system-assigned, which felt cumbersome.